src/lib/services/workflow-service.ts

```diff
@@ -293,30 +304,52 @@ export async function signalWorkflow({
   runId,
   signalName,
   signalInput,
+  settings,
+  accessToken,
 }: SignalWorkflowOptions) {
   const route = routeForApi('workflow.signal', {
     namespace,
     workflowId,
     runId,
   });
 
+  let payloads = null;
+
+  if (signalInput) {
+    if (settings?.codec?.endpoint) {
+      const awaitData = await convertPayloadsWithCodec({
+        payloads: { payloads: [JSON.parse(signalInput)] },
+        namespace,
+        settings,
+        accessToken,
+        encode: true,
+      });
+      if (get(lastDataEncoderStatus) === 'error') {
+        throw new Error(
+          get(lastDataEncoderError) || translate('encode-failed'),
+        );
+      }
```

Comment:
So the thing that tripped me up testing this and I'm sure users will run into is knowing the right structure to send payloads when a codec server is configured. 

Options when codec server is configured:
- Add a message to tell users to structure payload as `{ "metadata": { "encoding": "...." }, "data": "...." }`
- Automatically set the initial value of the input to `{ "metadata": { "encoding": "...." }, "data": "...." }` when signal modal is open


<img width="1728" alt="Screen Shot 2023-10-10 at 11 03 31 AM" src="https://github.com/temporalio/ui/assets/7967403/bdf803fe-9f6f-46d6-ae80-2a66b307c30b">

Reply:
Hmm yeah let me try both and see how it feels. Wish we could just do a placeholder here, but that would likely require it to be a `<textarea />` 🤔

Reply:
Could we try JSON.stringify({"metadata": {"encoding": "..."}, "data": "..."}, undefined, 2) to see if it renders it on multiple lines?

Reply:
Oh its just a placeholder. I would set it as the actual value so they can edit it

Reply:
Ah ok, there's a low likelihood of a user just be copy/pasting the input here then?

Reply:
They certainly could, but it's pretty easy and a common pattern to cmd-a and paste over the defaulted value. I think it would be really helpful to not have users need to enter these required keys.

<img width="1728" alt="Screen Shot 2023-10-10 at 11 03 31 AM" src="https://github.com/temporalio/ui/assets/7967403/1e27f210-d827-456c-b9bd-efbe0bf1fc31">

Reply:
Makes sense, updated ✅

Reply:
I might omit the `. . .` and just use empty strings but not gonna block the PR on it

Reply:
Yeah I was kinda thinking that too. probably best to do empty strings

Reply:
[Done](https://github.com/temporalio/ui/pull/1671/commits/d945dea86fa99152a6c0245ac7ca0b742a648995) ✅